packages/admin/admin/src/theme/createCometTheme.ts

```diff
@@ -57,11 +42,18 @@ export const createCometTheme = (
             },
         },
         shadows,
-        zIndex,
-        components: getComponentsTheme(passedComponentsOptions, { palette, typography, spacing, zIndex, shadows, breakpoints }),
+        zIndex: passedZIndexOptions,
         breakpoints: breakpointsOptions,
-    };
+    } satisfies ThemeOptions;
+
+    const combinedThemeOptionsBeforeAddingComponents = deepmerge(cometThemeOptionsBeforeAddingComponents, restPassedOptions);
+    const themeBeforeAddingComponents = createTheme(combinedThemeOptionsBeforeAddingComponents);
+
+    const cometThemeOptions = {
+        ...cometThemeOptionsBeforeAddingComponents,
+        components: getComponentsTheme(passedComponentsOptions, themeBeforeAddingComponents),
```

Comment:
Why is this special treatment for components necessary? 🤔

Reply:
Because the `styleOverrides` of some components require access to the theme (e.g. the colors or `spacing()`). 
That why we need to run `createTheme()` once and pass the result into `getComponentsTheme()`. 

Previously we only created the parts of the theme that were needed, but e.g. `createPalette()` can no longer be imported, due to: [Package layout updated](https://mui.com/material-ui/migration/upgrade-to-v7/#package-layout-updated). This was never officially supported apparently.